### PR TITLE
feat: calculator tool

### PR DESCRIPTION
**Pull request by michael-desmond**
Calculator tool based on [mathjs](https://mathjs.org/)

### CODE REVIEW

File: src/tools/calculator.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Only use the calculator tool if you need to perform a calculation.`;
+    inputSchema() {
+      return z.object({
+        expression: z.string().describe(`The mathematical expression to evaluate (e.g., "2 + 3 * 4"). Use Math.js basic expression syntax. Constants only.`),
+      });
+    }
+  
+    constructor({...options }: BaseToolOptions = {}) {
+      super(options);
+    }
```
Comment (Tomas2D) [inline src/tools/calculator.ts:39]: constructor not needed

File: src/tools/calculator.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    BaseToolOptions,
+    BaseToolRunOptions,
+    StringToolOutput,
+    Tool,
+    ToolInput,
+  } from "@/tools/base.js";
+  import { z } from "zod";
+  import { evaluate } from 'mathjs'
+  
+  export class CalculatorTool extends Tool<StringToolOutput, BaseToolOptions> {
```
Comment (Tomas2D) [inline src/tools/calculator.ts:27]: `BaseToolOptions` not needed

**Review by Tomas2D — CHANGES_REQUESTED** [review]
Great work! 🚀

Please rebase against main + add a simple test case.

File: src/tools/calculator.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  
+    // extra (has functional impact)
+    'evaluate':   function () { throw new Error('Function evaluate is disabled') },
+    'parse':      function () { throw new Error('Function parse is disabled') },
+    'simplify':   function () { throw new Error('Function simplify is disabled') },
+    'derivative': function () { throw new Error('Function derivative is disabled') },
+    'resolve':    function () { throw new Error('Function resolve is disabled') },
+  }, { override: true })
+
+  export class CalculatorTool extends Tool<StringToolOutput> {
```
Comment (matoushavlena) [inline src/tools/calculator.ts:42]: I would include a disclaimer/warning that are are potential security and stability risks involved in using `.evaluate`, even with the limited version. Pointing to the mathjs [documentation](https://mathjs.org/docs/expressions/security.html).

@Tomas2D This is likely not the first time we would be introducing a potentially dangerous tool. Any specific guidance on how approach this (comment at the top, having a specific flag turned on/off)? For example check the request tool from [langchain.](https://github.com/langchain-ai/langchain/blob/34fc00aff17d1913a1cfe4b205e732982f98f7b8/libs/community/langchain_community/tools/requests/tool.py#L36). I think users should know that this is not safe tool unless run in isolation. Another approach could be to add `unsafe` suffix to the tool name for all tools we cannot guarantee safeness OOTB.

**Review by matoushavlena — COMMENTED** [review]
@Tomas2D Is there any guidance on using Web Workers for tools that might have impact on CPU/memory?

File: src/tools/calculator.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    'createUnit': function () { throw new Error('Function createUnit is disabled') },
+    'reviver':    function () { throw new Error('Function reviver is disabled') },
+  
+    // extra (has functional impact)
+    'evaluate':   function () { throw new Error('Function evaluate is disabled') },
+    'parse':      function () { throw new Error('Function parse is disabled') },
+    'simplify':   function () { throw new Error('Function simplify is disabled') },
+    'derivative': function () { throw new Error('Function derivative is disabled') },
+    'resolve':    function () { throw new Error('Function resolve is disabled') },
+  }, { override: true })
```
Comment (Tomas2D) [inline src/tools/calculator.ts:40]: From my point of view, this should all be in the class constructor with the ability to pass the user's custom config (initialization).

File: src/tools/calculator.ts (hunk 3 above)
Comment (Tomas2D) [reply src/tools/calculator.ts:42]: I agree that there should be a disclaimer.

Comment (Tomas2D) [reply src/tools/calculator.ts:42]: If we decide to use the flag, we would need to add it to other tools, like `WebCrawlerTool`.

File: src/tools/calculator.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    name = "Calculator";
+    description = `A calculator tool that performs basic arithmetic operations like addition, subtraction, multiplication, and division. 
+Only use the calculator tool if you need to perform a calculation.`;
+    inputSchema() {
+      return z.object({
+        expression: z.string().describe(`The mathematical expression to evaluate (e.g., "2 + 3 * 4"). Use Mathjs basic expression syntax. Constants only.`),
+      });
+    }
+  
+    protected async _run({ expression }: ToolInput<this>, options?: BaseToolRunOptions) {
```
Comment (Tomas2D) [inline src/tools/calculator.ts:52]: remove the `options` parameter as it is not used

**Review by Tomas2D — CHANGES_REQUESTED** [review]

File: src/tools/calculator.ts (hunk 3 above)
Comment (michael-desmond) [reply src/tools/calculator.ts]: I added a warning re. security and stability of the tool, linking to the mathjs documentation.

File: src/tools/calculator.ts (hunk 4 above)
Comment (michael-desmond) [reply src/tools/calculator.ts]: I moved this to the class constructor. I wasn't quite sure what you mean by users custom config, perhaps you can say more on that.

File: src/tools/calculator.ts (hunk 5 above)
Comment (michael-desmond) [reply src/tools/calculator.ts]: Removed.

File: src/tools/calculator.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+ */
+
+import {
+    BaseToolOptions,
+    StringToolOutput,
+    Tool,
+    ToolInput,
+  } from "@/tools/base.js";
+  import { z } from "zod";
+  import { create, all, MathJsInstance } from 'mathjs'
```
Comment (Tomas2D) [inline src/tools/calculator.ts:24]: 'MathJsInstance' is defined but never used

File: src/tools/calculator.ts [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+   * https://mathjs.org/docs/expressions/security.html before using this tool. 
+   */
+  export class CalculatorTool extends Tool<StringToolOutput> {
+    name = "Calculator";
+    description = `A calculator tool that performs basic arithmetic operations like addition, subtraction, multiplication, and division. 
+Only use the calculator tool if you need to perform a calculation.`;
+
+    inputSchema() {
+      return z.object({
+        expression: z.string().describe(`The mathematical expression to evaluate (e.g., "2 + 3 * 4"). Use Mathjs basic expression syntax. Constants only.`),
```
Comment (Tomas2D) [inline src/tools/calculator.ts:40]: change to 
```
z.string().min(1).describe(`The mathematical expression to evaluate (e.g., "2 + 3 * 4"). Use Mathjs basic expression syntax. Constants only.`)
```

File: src/tools/calculator.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      });
+    }
+
+    protected limitedEvaluate: any;
```
Comment (Tomas2D) [inline src/tools/calculator.ts:44]: `protected limitedEvaluate: typeof math.evaluate;`

**Review by Tomas2D — CHANGES_REQUESTED** [review]

File: src/tools/calculator.ts [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        'reviver':    function () { throw new Error('Function reviver is disabled') },
+      
+        // extra (has functional impact)
+        'evaluate':   function () { throw new Error('Function evaluate is disabled') },
+        'parse':      function () { throw new Error('Function parse is disabled') },
+        'simplify':   function () { throw new Error('Function simplify is disabled') },
+        'derivative': function () { throw new Error('Function derivative is disabled') },
+        'resolve':    function () { throw new Error('Function resolve is disabled') },
+      }, { override: true })
+    }
```
Comment (Tomas2D) [inline src/tools/calculator.ts:64]: I was thinking about something like this.

```

export interface CalculatorToolInput {
   config?: ConfigOptions
   imports?: {
      entries: ImportObject | ImportObject[],
      options?: ImportOptions
   }
}

...

    constructor({ config, imports ...options }: CalculatorToolInput = {}) {
      super(options);
      const math = create(all, config)
      this.limitedEvaluate = math.evaluate
      // Disable use potentially vulnerable functions
      math.import({
        // most important (hardly any functional impact)
        'import':     function () { throw new Error('Function import is disabled') },
        'createUnit': function () { throw new Error('Function createUnit is disabled') },
        'reviver':    function () { throw new Error('Function reviver is disabled') },

        // extra (has functional impact)
        'evaluate':   function () { throw new Error('Function evaluate is disabled') },
        'parse':      function () { throw new Error('Function parse is disabled') },
        'simplify':   function () { throw new Error('Function simplify is disabled') },
        'derivative': function () { throw new Error('Function derivative is disabled') },
        'resolve':    function () { throw new Error('Function resolve is disabled') },
         ...imports?.entries,
      }, { override: true, ...imports?.options })
    }

...
```

**Review by Tomas2D — CHANGES_REQUESTED** [review]
CI/CD fails because of invalid formatting. Run `yarn format:fix` to resolve that issue. Then, I will merge it.

**Review by Tomas2D — APPROVED** [review]

**Pull request merged**